contrib/oidc/playbooks/oidc_configure.yml

```diff
@@ -0,0 +1,428 @@
+        mode: '0600'
+
+    - name: Generate CSR
+      openssl_csr:
+        path: "{{ dex_tls_dir }}/tls.csr"
+        privatekey_path: "{{ dex_tls_dir }}/tls.key"
+        common_name: "{{ inventory_hostname }}"
+        subject_alt_name: "DNS:{{ inventory_hostname }}"
+        owner: "{{ dex_user }}"
+        group: "{{ dex_user }}"
+        mode: '0644'
+
+    - name: Generate self-signed certificate
+      openssl_certificate:
+        path: "{{ dex_tls_dir }}/tls.crt"
+        privatekey_path: "{{ dex_tls_dir }}/tls.key"
+        csr_path: "{{ dex_tls_dir }}/tls.csr"
+        provider: selfsigned
+        selfsigned_not_after: "+365d"
+        owner: "{{ dex_user }}"
+        group: "{{ dex_user }}"
+        mode: '0644'
+
+    - name: Create Dex config file
+      copy:
+        dest: "{{ dex_config_dir }}/dex.yaml"
+        content: |
+          issuer: https://{{ inventory_hostname }}
+          storage:
+            type: sqlite3
+            config:
+              file: /var/dex/data.db
+          web:
+            http: 0.0.0.0:5556
+            https: 0.0.0.0:5554
+            tlsCert: /etc/dex/tls/tls.crt
+            tlsKey: /etc/dex/tls/tls.key
+          connectors:
+          - type: ldap
+            name: OpenLDAP
+            id: ldap
+            config:
+              host: {{ inventory_hostname }}:389
+              insecureNoSSL: true
+              bindDN: cn=admin,{{ ldap_base_dn }}
+              bindPW: {{ ldap_admin_password }}
+              userSearch:
+                baseDN: ou=users,{{ ldap_base_dn }}
+                filter: "(objectClass=inetOrgPerson)"
+                username: uid
+                idAttr: uid
+                emailAttr: mail
+                nameAttr: cn
+                preferredUsernameAttr: uid
+              groupSearch:
+                baseDN: ou=groups,{{ ldap_base_dn }}
+                filter: "(objectClass=groupOfNames)"
+                userAttr: DN
+                groupAttr: member
+                nameAttr: cn
+          oauth2:
+            skipApprovalScreen: true
+          staticClients:
+          - id: openshift-console
+            redirectURIs:
+            - https://console-openshift.example.com/oauth2/callback
+            name: OpenShift Console
+            secret: secret123
+        owner: "{{ dex_user }}"
+        group: "{{ dex_user }}"
+        mode: '0644'
+      notify: restart dex
+
+    - name: Run Dex container
+      containers.podman.podman_container:
+        name: dex
+        image: ghcr.io/dexidp/dex
+        command: ["dex", "serve", "/etc/dex/config/dex.yaml"]
+        volumes:
+          - "{{ dex_config_dir }}:/etc/dex/config:ro,Z"
+          - "{{ dex_tls_dir }}:/etc/dex/tls:ro,Z"
+          - "{{ dex_dir }}/data:/var/dex:rw,Z"
+        ports:
+          - "80:5556"
+          - "443:5554"
+        user: "{{ dex_uid }}:{{ dex_uid }}"
+        # network: host
+        state: started
+
+    - name: Wait for Dex container to start
```

Comment:
Is this because this is the only container in the instance? Would it be more robust if you wait for a health check URL to succeed?

Reply:
good catch. fixed